libraries/AP_Scripting/generator/src/main.c

```diff
@@ -369,14 +375,26 @@ struct userdata {
 static struct userdata *parsed_userdata;
 static struct userdata *parsed_ap_objects;
 
-void sanatize_name(char **dest, char *src) {
-  *dest = (char *)allocate(strlen(src) + 1);
-  strcpy(*dest, src);
-  char *position = strchr(*dest, ':');
+void sanatize_character(char **str, char character) {
+  char *position = strchr(*str, character);
   while (position) {
     *position = '_';
-    position = strchr(position, ':');
+    position = strchr(position, character);
   }
+}
+
+void sanatize_name(char **dest, char *src) {
```

Comment:
"sanatize" should be "sanitise" or "sanitize".  we normally use the british english spelling which would be "sanitise" but either is ok.

Reply:
that is a existing issue, I have fixed the name on the new `sanitize_character` function.